toolchain/parse/typed_nodes_test.cpp

```diff
@@ -276,6 +276,113 @@ Aggregate [^:]*: success
 )Trace"));
 }
 
+TEST_F(TypedNodeTest, Token) {
+  auto* tree = &GetTree(R"carbon(
+    var n: i32 = 0;
+  )carbon");
+  auto file = tree->ExtractFile();
+
+  ASSERT_EQ(file.decls.size(), 1);
+
+  auto n_var = tree->ExtractAs<VariableDecl>(file.decls[0]);
+  ASSERT_TRUE(n_var.has_value());
+  EXPECT_TRUE(n_var->token.index.index == 7);
```

Comment:
FWIW, I'd actually expect this test to be one of the least brittle ones in this file: the tests that check the verification trace output elsewhere in this file are substantially more brittle, and break frequently when the tree structure changes.

But, switched to check the token kind instead of the token index, to avoid making assumptions about the meaning of token indexes.